clients/src/main/java/org/apache/kafka/common/requests/FetchResponse.java

```diff
@@ -143,15 +143,29 @@ public FetchResponse(Struct struct) {
     public Send toSend(String dest, RequestHeader requestHeader) {
         ResponseHeader responseHeader = new ResponseHeader(requestHeader.correlationId());
 
+        boolean hasThrottleTime = struct.hasField(THROTTLE_TIME_KEY_NAME);
```

Comment:
It doesn't in one sense (it is still just a conditional on `hasField(THROTTLE_TIME_KEY_NAME)`), but it does in the sense that previously we had 2 blocks, one for each version, each of which contained every single field for that message version. That approach makes it really easy to match up the code to the protocol specs in KIPs.

Again, not necessarily an issue, just wanted to make sure we've considered challenges in maintaining the code moving forward.